config/confighttp/confighttp_test.go

```diff
@@ -255,6 +249,7 @@ func TestHTTPClientSettingWithAuthConfig(t *testing.T) {
 		shouldErr    bool
 		settings     HTTPClientSettings
 		extensionMap map[config.ComponentID]component.Extension
+		host         component.Host
```

Comment:
No cases are setting these. Should it really be added here?

Reply:
https://github.com/open-telemetry/opentelemetry-collector/pull/5083/files#diff-ae54372cbf36c6ca3ce262dc39d7db67e1d71bba20c8838a9906bc123133c8b6R312
Here its being used.

Reply:
It's consumed there, but no tests are creating a custom value for it. Each test is defining a map of components, but that's not being used anymore. Shouldn't you create a host based on that map?

Reply:
@jpkrohling  let me walk through it again to end to end flow. I will create host based on that map accordingly.

Reply:
It is not used anywhere.